### PR TITLE
Updates to pr1827

### PR DESCRIPTION
**Pull request by p-bakker**
Fixes issues introduced in #1827, namely:
- runner_support.js mixing usage of the `print` and `console.log` commands, neither of which will both work on any environment
- exceptions occurring during running of tests cluttering the stdout

Includes extracting `createIterableHelper` out from runner_support, as it's used by the `node.js` script, which must remain working on older NodeJS versions (and thus should be authored in ES3), while runner_support.js does not have that ES3 requirement  (and thus can use newer syntax)

### CODE REVIEW

File: .gitignore
Comment (ljharb) [inline .gitignore]: i'm not sure what these are, but specific tools' files should be in your global gitconfig, not added to every project you touch

File: test-utils/testHelpers.js
Comment (ljharb) [inline test-utils/testHelpers.js:1]: this ideally would be enforced using eslintrc and "overrides" - is there a reason this *must* be a string? could it be an actual JS function, thus lintable, and when concatenated into a string elsewhere, use Function.prototype.toString?

Comment (p-bakker) [reply test-utils/testHelpers.js:1]: If we don't store it as a string, we're relying on the (proper) implementation of Function.prototype.toString() in the executing environment, whatever that may me.

I cannot judge whether it's safe to rely on that... thoughts?

File: .gitignore
Comment (p-bakker) [reply .gitignore]: .project is from Eclipse and it's required to have any project in your workspace. In this case it can be omitted, as there's no relevant info in it, but that is not always the case, so it cannot be excluded globally, I guess similarly to `.idea/`?

.history is a VS Code (plugin?) thing which I think makes sense to exclude globally, will check

File: test-utils/testHelpers.js
Comment (ljharb) [reply test-utils/testHelpers.js:1]: yeah, that's a fair point :-/ probably best to keep it as a string just in case.

File: .gitignore
Comment (ljharb) [reply .gitignore]: `.idea/` is just a default from github, it shouldn't be there either.

why can't `.project` be excluded in your global `.gitignore`?

Comment (p-bakker) [reply .gitignore]: See https://einverne.github.io/post/2016/04/should-i-git-ignore-project-file-under-eclipse.html

Basically: for many projects in Eclipse you want the .project file to be committed, hence it shouldn't be excluded globally

Should I remove `.idea/` then?

Comment (ljharb) [reply .gitignore]: That seems like an unfortunate design in Eclipse :-/

sure, let's remove `.idea/`

File: runner_support.js
Comment (ljharb) [inline runner_support.js]: `||` would work fine here, ftr

File: runner_support.js
Comment (ljharb) [inline runner_support.js]: let's keep the curly braces; don't want another gotofail :-)
```suggestion
                            if (expression) {
                                ${logCommand}("[SUCCESS]");
                            }
```

File: runner_support.js
Comment (ljharb) [inline runner_support.js]:
```suggestion
                            if (eval(testCode)) {
                                ${logCommand}("[SUCCESS]");
                            }
```

File: runner_support.js
Comment (ljharb) [inline runner_support.js]:
```suggestion
                        } catch (e) {}
```

**Review by ljharb — COMMENTED** [review]
It seems like you're adding try/catches in a number of places that didn't have them before; won't that mask errors in the test code we want caught?

**Comment by p-bakker** [comment]
> It seems like you're adding try/catches in a number of places that didn't have them before; won't that mask errors in the test code we want caught?

Prior to #1827 there also were try/catched around the eval of the testcode:
- https://github.com/compat-table/compat-table/pull/1827/files#diff-9ae6f9d9f0960230222e4bfb5f9282bcb02bd4e8e4220272bdfe4bd7ef1db3a7L88
- https://github.com/compat-table/compat-table/pull/1827/files#diff-48b230a7f01f4595f2a9df8af1db79063c8a0e0458de67e4873877d98880cbb6L83
- https://github.com/compat-table/compat-table/pull/1827/files#diff-3744f0f40c1d8d08319f4dce83919405e334cf138b46cc16e9848cb88a1a5489L159
- etc.

File: .gitignore
Comment (p-bakker) [reply .gitignore]: done

File: runner_support.js
Comment (p-bakker) [reply runner_support.js]: replaced

File: runner_support.js
Comment (p-bakker) [reply runner_support.js]: done

File: runner_support.js
Comment (p-bakker) [reply runner_support.js]: done

File: runner_support.js
Comment (p-bakker) [reply runner_support.js]: done

File: .gitignore
Comment (ljharb) [reply .gitignore]: `.project` still should be removed, though, despite the inconvenience for yourself.

**Comment by ljharb** [comment]
gotcha, so it's restoring them, good.

File: .gitignore
Comment (p-bakker) [reply .gitignore]: Done

**Review by ljharb — APPROVED** [review]
LGTM, but i'll wait for another approval before i merge

**Comment by p-bakker** [comment]
I see the pretest stage fails due to linting. For now I fixed it to the specific file that had warnings, but going forward what's the desired way?

1. Globally allow newer js, limit to es3 only on the data-*.js files, node.js en master.js? Or
2. The other way around: globally limit to es3, allowing newer js on runner_support.js/build.js & the various runner .js files?

**Comment by p-bakker** [comment]
Is there anything I must do now to move this PR along?

**Comment by ljharb** [comment]
> I see the pretest stage fails due to linting. For now I fixed it to the specific file that had warnings, but going forward what's the desired way?
> 
> 1. Globally allow newer js, limit to es3 only on the data-*.js files, node.js en master.js? Or
> 2. The other way around: globally limit to es3, allowing newer js on runner_support.js/build.js & the various runner .js files?

i would go with option 2; the default should be the safest thing.

**Comment by p-bakker** [comment]
> i would go with option 2; the default should be the safest thing.

k, that's what I figured as well and thus what I already did :-)

Just now waiting for that second approval...

**Comment by p-bakker** [comment]
@chicoxyzzy any chance you can review this one?

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**
